androidbrowserhelper/src/main/java/com/google/androidbrowserhelper/trusted/LocationProviderAndroid.java

```diff
@@ -0,0 +1,206 @@
+// Copyright 2020 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.androidbrowserhelper.trusted;
+
+import android.content.Context;
+import android.location.Criteria;
+import android.location.Location;
+import android.location.LocationListener;
+import android.location.LocationManager;
+import android.os.Bundle;
+import android.os.Looper;
+import android.os.RemoteException;
+import android.util.Log;
+
+import java.util.List;
+
+import androidx.browser.trusted.TrustedWebActivityCallbackRemote;
+
+/**
+ * This is a LocationProvider using Android APIs [1]. It is a separate class for clarity
+ * so that it can manage all processing completely on the UI thread. The container class
+ * ensures that the start/stop calls into this class are done on the UI thread.
+ * <p>
+ * [1] https://developer.android.com/reference/android/location/package-summary.html
+ */
+public class LocationProviderAndroid implements LocationListener {
+    private static final String TAG = "TWA_LocationProvider";
+
+    private static final String EXTRA_NEW_LOCATION_AVAILABLE_CALLBACK = "onNewLocationAvailable";
+    private static final String EXTRA_NEW_LOCATION_ERROR_CALLBACK= "onNewLocationError";
+
+    private LocationManager mLocationManager;
+    private boolean mIsRunning;
+    private TrustedWebActivityCallbackRemote mCallback;
+    private static LocationProviderAndroid sProvider;
+
+    public static LocationProviderAndroid getInstance() {
+        if (sProvider == null) {
+            sProvider = new LocationProviderAndroid();
+        }
+        return sProvider;
+    }
+
+    LocationProviderAndroid() {
```

Comment:
You don't need this - an empty, no parameter constructor is created automatically for you.